diagnostic_updater/diagnostic_updater/_diagnostic_updater.py

```diff
@@ -305,36 +324,41 @@ def setHardwareID(self, hwid):
 
     def _check_diagnostic_period(self):
         """Recheck the diagnostic_period on the parameter server."""
-
         # This was getParamCached() call in the cpp code. i.e. it would throttle
         # the actual call to the parameter server using a notification of change
         # mechanism.
         # This is not available in rospy. Hence I throttle the call to the
         # parameter server using a standard timeout mechanism (4Hz)
-
-        now = rospy.Time.now()
-        if  now >= self.last_time_period_checked + rospy.Duration(0.25):
+        clock = Clock(clock_type=ClockType.STEADY_TIME)
+        now = clock.now()
+        if now >= self.last_time_period_checked:
             try:
-                self.period = rospy.get_param("~diagnostic_period", 1)
                 self.last_time_period_checked = now
-            except (httplib.CannotSendRequest, httplib.ResponseNotReady):
+            except (httplib2.CannotSendRequest, httplib2.ResponseNotReady):
```

Comment:
Since the param call here has been removed, it looks like the try-except can be removed here too, and the dependency on httplib2 can be removed completely. (This should also fix the CI failure).

Reply:
Yeah; I think this was intended to catch errors if the parameter retrieval failed. I'm familiar with ROS2, but it seems like this should be replaced by a more appropriate `rclpy` exception (or completely removed, if param retrieval cannot fail in `rclpy`)

Reply:
So I removed the exception handling all together as I believe the `rclpy` implementation doesn't throw any errors:
https://github.com/ros2/rclpy/blob/master/rclpy/rclpy/node.py#L231-L240

The only thing which could happen is that the parameter is not set. However, as we initialize the parameter ourselves in the constructor, this can't happen.

Reply:
Great!